--- git/bfd/configureOld	2014-12-26 21:21:54.977919000 +0000
+++ git/bfd/configure	2014-12-26 22:50:00.274708939 +0000
@@ -12302,8 +12302,8 @@
   bfdlibdir='$(libdir)'
   bfdincludedir='$(includedir)'
   if test "${host}" != "${target}"; then
-    bfdlibdir='$(exec_prefix)/$(host_noncanonical)/$(target_noncanonical)/lib'
-    bfdincludedir='$(exec_prefix)/$(host_noncanonical)/$(target_noncanonical)/include'
+    bfdlibdir='$(exec_prefix)/lib/$(target_noncanonical)'
+    bfdincludedir='$(exec_prefix)/include/$(target_noncanonical)'
   fi
 
 
