diff --git a/opcodes/configure b/opcodes/configure
index 64c30c7..5aef940 100755
--- a/opcodes/configure
+++ b/opcodes/configure
@@ -12357,8 +12357,8 @@ fi
   bfdlibdir='$(libdir)'
   bfdincludedir='$(includedir)'
   if test "${host}" != "${target}"; then
-    bfdlibdir='$(exec_prefix)/$(host_noncanonical)/$(target_noncanonical)/lib'
-    bfdincludedir='$(exec_prefix)/$(host_noncanonical)/$(target_noncanonical)/include'
+    bfdlibdir='$(exec_prefix)/lib/$(target_noncanonical)'
+    bfdincludedir='$(exec_prefix)/include/$(target_noncanonical)'
   fi
 
 
